lib/ansible/modules/blockinfile.py

```diff
@@ -287,12 +297,16 @@ def main():
     else:
         insertre = None
 
+    wrap_empty_line = b("")
+    if module.boolean(params['wrap_with_blank_lines']):
+        wrap_empty_line = b(os.linesep)
+
     marker0 = re.sub(b(r'{mark}'), b(params['marker_begin']), marker) + b(os.linesep)
     marker1 = re.sub(b(r'{mark}'), b(params['marker_end']), marker) + b(os.linesep)
     if present and block:
         if not block.endswith(b(os.linesep)):
             block += b(os.linesep)
-        blocklines = [marker0] + block.splitlines(True) + [marker1]
+        blocklines = [wrap_empty_line] + [marker0] + block.splitlines(True) + [marker1] + [wrap_empty_line]
```

Comment:
This will have the side effect of reinserting new blank lines every time the task runs even when no change needs to be made. The new tests should focus on `wrap_with_blank_lines: True` since the pre-existing tests cover the default case. They should test
* adding a block with newlines (covered)
* adding newlines to a pre-existing block without newlines
* that repeat operations don't make additional changes

Reply:
Absolutely true, I will adjust the behavior to ensure the empty lines are not added if the block is already present and I will add the two tests mentioned above. Thank you for the valuable feedback, I will work on it as quickly as possible